metagraph/tests/annotation/test_annotated_dbg_helpers.cpp

```diff
@@ -7,65 +7,184 @@
+                    load_coords(std::move(dynamic_cast<ColumnCompressed<>&>(*anno_graph->annotator_)),
+                                std::vector<std::string>{
+                                    out_path + ColumnCompressed<>::kExtension
+                                })
                 ))
-            )
-        );
+            );
+        }
+    } else {
+        for (size_t i = 0; i < sequences.size(); ++i) {
+            anno_graph->annotate_sequence(std::string(sequences[i]), { labels[i] });
+        }
+
+        if constexpr(std::is_same_v<Annotation, ColumnCompressed<>>)
+            return anno_graph;
+
+        if constexpr(!std::is_same_v<Annotation, RowDiffColumnAnnotator>
+                && !std::is_same_v<Annotation, RowDiffCoordAnnotator>) {
+            return std::make_unique<AnnotatedDBG>(
+                graph,
+                std::unique_ptr<AnnotatedDBG::Annotator>(
+                    convert<Annotation>(
+                        std::move(dynamic_cast<ColumnCompressed<>&>(*anno_graph->annotator_))
+                    )
+                )
+            );
+        }
+    }
 
-    return anno_graph;
+    if constexpr(std::is_same_v<Annotation, RowDiffCoordAnnotator>
+            || std::is_same_v<Annotation, RowDiffColumnAnnotator>) {
+        static_assert(std::is_same_v<Graph, DBGSuccinct>);
+        size_t num_columns = anno_graph->get_annotator().get_label_encoder().size();
+        std::filesystem::path tmp_dir = utils::create_temp_dir("", "test_col");
+        std::string out_path = tmp_dir/"test_col";
+        anno_graph->get_annotator().serialize(out_path);
+        std::vector<std::string> files { out_path + ColumnCompressed<>::kExtension };
+        base_graph->serialize(tmp_dir/"test_col");
+        std::string graph_fname = std::string(tmp_dir/"test_col") + base_graph->file_extension();
+        convert_to_row_diff(files, graph_fname, 10e9, 100, tmp_dir, tmp_dir,
+                            static_cast<annot::RowDiffStage>(0),
+                            "", false, coordinates);
+        convert_to_row_diff(files, graph_fname, 10e9, 100, tmp_dir, tmp_dir,
+                            static_cast<annot::RowDiffStage>(1),
+                            "", false, coordinates);
+        convert_to_row_diff(files, graph_fname, 10e9, 100, tmp_dir, tmp_dir,
+                            static_cast<annot::RowDiffStage>(2),
+                            "", false, coordinates);
+        const std::string anchors_file = graph_fname + annot::binmat::kRowDiffAnchorExt;
+        const std::string fork_succ_file = graph_fname + annot::binmat::kRowDiffForkSuccExt;
+        if (!std::filesystem::exists(anchors_file)) {
+            logger->error("Anchor bitmap {} does not exist.", anchors_file);
+            std::exit(1);
+        }
+        if (!std::filesystem::exists(fork_succ_file)) {
+            logger->error("Fork successor bitmap {} does not exist", fork_succ_file);
+            std::exit(1);
+        }
+
```

Comment:
I would look closely and double-check this function. The row-diff converters are tested pretty well, so I doubt there is any bug there.

Reply:
Do you have any ideas about what could be wrong? I've made a few tweaks (explicitly setting the `row_count` and `row_reduction` file extensions, increasing the amount of available RAM, tweaking the max path length, setting separate output and swap dirs), but none of these seem to fix things.

Reply:
Does this graph have a mask for dummy k-mers?

Reply:
It does, but removing the mask doesn't seem to fix the issue.